api/v1beta2/ocirepository_types.go

```diff
@@ -203,6 +203,16 @@ type OCIRepositoryStatus struct {
 	// +optional
 	Artifact *Artifact `json:"artifact,omitempty"`
 
+	// ObservedIgnore is the observed exclusion patterns used for constructing
+	// the source artifact.
+	// +optional
+	ObservedIgnore *string `json:"observedIgnore,omitempty"`
```

Comment:
This appears like this:

```yaml
  observedGeneration: 10
  observedIgnore: |
    hpa.yaml
    foo.txt
  observedLayerSelector:
    mediaType: application/vnd.docker.image.rootfs.diff.tar.gzip
    operation: extract
  url: http://source-controller.flux-system.svc.cluster.local./ocirepository/default/podinfo-layer/latest.tar.gz
```

Not sure if it's okay or if we should convert it to a string containing the checksum of the ignore valye.

Reply:
Let’s go with https://github.com/fluxcd/source-controller/blob/main/api/v1beta2/gitrepository_types.go#L228

Reply:
There are disadvantages to this:

1. The user now needs to be aware of the mechanics of `.generation` and `.status.observedGeneration` to be able to assume that the checksum will match the current `.spec` elements, over just looking at a reflection of plain values.
2. By combining the values into a single checksum, you lose the ability to selectively notice changes to a sub-set of the spec.

Reply:
We'll end up with 99% of the spec reflected into the status sub-resource which feels very wrong to me. IMO we should be reconciling on generation drift, but in SC we decided to skip reconciliation when `interval` changes, which got us to this point where we have to track every other filed. I'm for triggering a reconciliation on spec changes, is not like interval will change so often.

Reply:
I"ve updated the implementation to add content config checksum in the status. It looks like the following:
```yaml
  contentConfigChecksum: sha256:e5be314a56171026ffe3b828fec184d6144fb602511ee7243d02193e26114c45
  observedGeneration: 6
  url: http://source-controller.flux-system.svc.cluster.local./ocirepository/default/podinfo/latest.tar.gz
```

Based on some private discussions, we are adding content config checksum here for now to align with the GitRepository API.
In a future release, we'll replace this checksums with more explicit observed values in the status to improve the usability by looking at the status and easily seeing the differences in the declared state and the observations made by the controller.